src/components/collection/ResourceConfigForm.tsx

```diff
@@ -50,10 +50,40 @@ function ResourceConfigForm({ collectionName, readOnly = false }: Props) {
         },
     };
 
+    // Find field with x-collection-name annotation
+    const collectionNameFieldKey = useMemo(() => {
+        if (resourceSchema.properties) {
+            // Find the field with the collection name annotation
+            const collectionNameField =
+                Object.entries(resourceSchema.properties).find(
+                    ([_, value]) =>
+                        value && Annotations.defaultResourceConfigName in value
+                ) ?? [];
+
+            // Try to fetch the key
+            return collectionNameField[0];
+        }
+
+        return null;
+    }, [resourceSchema.properties]);
```

Comment:
This chunk isn't _too_ heavy but since we don't need to recalculate it often decided to move this into a stand alone memo.